client/asset/ltc/spv.go

```diff
@@ -0,0 +1,1049 @@
+
+func (s *spvService) GetBlockHash(height int64) (*chainhash.Hash, error) {
+	ltcHash, err := s.ChainService.GetBlockHash(height)
+	if err != nil {
+		return nil, err
+	}
+	return convertHashToBTC(*ltcHash), nil
+}
+
+func (s *spvService) BestBlock() (*headerfs.BlockStamp, error) {
+	bs, err := s.ChainService.BestBlock()
+	if err != nil {
+		return nil, err
+	}
+	return &headerfs.BlockStamp{
+		Height:    bs.Height,
+		Hash:      *convertHashToBTC(bs.Hash),
+		Timestamp: bs.Timestamp,
+	}, nil
+}
+
+func (s *spvService) Peers() []btc.SPVPeer {
+	rawPeers := s.ChainService.Peers()
+	peers := make([]btc.SPVPeer, len(rawPeers))
+	for i, p := range rawPeers {
+		peers[i] = p
+	}
+	return peers
+}
+
+func (s *spvService) GetBlockHeight(h *chainhash.Hash) (int32, error) {
+	return s.ChainService.GetBlockHeight(convertHashToLTC(*h))
+}
+
+func (s *spvService) GetBlockHeader(h *chainhash.Hash) (*wire.BlockHeader, error) {
+	hdr, err := s.ChainService.GetBlockHeader(convertHashToLTC(*h))
+	if err != nil {
+		return nil, err
+	}
+	return &wire.BlockHeader{
+		Version:    hdr.Version,
+		PrevBlock:  *convertHashToBTC(hdr.PrevBlock),
+		MerkleRoot: *convertHashToBTC(hdr.MerkleRoot),
+		Timestamp:  hdr.Timestamp,
+		Bits:       hdr.Bits,
+		Nonce:      hdr.Nonce,
+	}, nil
+}
+
+func (s *spvService) GetCFilter(blockHash chainhash.Hash, filterType wire.FilterType, _ ...btcneutrino.QueryOption) (*gcs.Filter, error) {
+	f, err := s.ChainService.GetCFilter(*convertHashToLTC(blockHash), ltcwire.GCSFilterRegular)
+	if err != nil {
+		return nil, err
+	}
+
+	b, err := f.Bytes()
+	if err != nil {
+		return nil, err
+	}
+
+	return gcs.FromBytes(f.N(), f.P(), DefaultM, b)
+}
+
+func (s *spvService) GetBlock(blockHash chainhash.Hash, _ ...btcneutrino.QueryOption) (*btcutil.Block, error) {
+	blk, err := s.ChainService.GetBlock(*convertHashToLTC(blockHash))
+	if err != nil {
+		return nil, err
+	}
+
+	b, err := blk.Bytes()
+	if err != nil {
+		return nil, err
+	}
+
+	return btcutil.NewBlockFromBytes(b)
+}
+
+func convertHashToBTC(src ltcchainhash.Hash) *chainhash.Hash {
+	var tgt chainhash.Hash
+	copy(tgt[:], src[:])
+	return &tgt
+}
+
+func convertHashToLTC(src chainhash.Hash) *ltcchainhash.Hash {
+	var tgt ltcchainhash.Hash
+	copy(tgt[:], src[:])
+	return &tgt
+}
+
+func convertMsgTxToBTC(tx *ltcwire.MsgTx) (*wire.MsgTx, error) {
+	buf := new(bytes.Buffer)
+	if err := tx.Serialize(buf); err != nil {
+		return nil, err
+	}
+
+	btcTx := new(wire.MsgTx)
+	if err := btcTx.Deserialize(buf); err != nil {
+		return nil, err
+	}
+	return btcTx, nil
+}
+
+func convertMsgTxToLTC(tx *wire.MsgTx) (*ltcwire.MsgTx, error) {
+	buf := new(bytes.Buffer)
+	if err := tx.Serialize(buf); err != nil {
+		return nil, err
+	}
+	ltcTx := new(ltcwire.MsgTx)
+	if err := ltcTx.Deserialize(buf); err != nil {
+		return nil, err
+	}
+
+	// 153280a7351c702ce4e8d87a878cc3d1eab41650ce0b895a3598b62c3778ad5f from peer=2 was not accepted: non-mandatory-script-verify-flag (Witness program hash mismatch)
```

Comment:
Something you hit that was a concern?  Figure it out?

Reply:
Yep, figured it out.